### ISSUE TITLE
Include files from arbitrary locations (like buildDir/work)

### ISSUE THREAD

**Issue by mglazer** [palantir/gradle-java-distribution#132]
We have a service which generates some 100 or so files automatically which are included in the output distribution.

At the moment, this task dictates that we must store these generated files in `projectDir/service`. 

It would be nice if we could instead supply a path to an alternate location to pull files from, like `buildDir/work`. Otherwise, we're cluttering our git repo with auto-generated files (or at the very least leaving behind a dirty git repo).

**Comment by bmoylan** [comment]
I agree this could be better, the workaround we landed on is pretty detailed .gitignore rules for service. For example service/web is ignored and we copy web assets in there pre-archive but service/bin is not.

**Comment by mglazer** [comment]
Mind if I submit a PR to make this moderately more configurable? Basically, allow extraDirs (name up for discussion), which just adds to projectDir and pulls in everything from those directories as well.

**Comment by bmoylan** [comment]
Can this be done by just adding things like

```
distTar.from("$buildDir/work") {
  into "$baseName/service/extraStuff"
}
```

**Comment by mglazer** [comment]
My original comment said this worked. That was before I did a full verification of the output. It doesn't work.

The Gradle Distribution task:

https://docs.gradle.org/current/userguide/distribution_plugin.html

Creates a Distribution object that enables you to configure the distribution contents. I'd assume we'd have to do a similar idiom here.

**Comment by uschi2000** [comment]
I'm pretty sure I have configured the distribution before somewhere. Mike, can you share a repo where you're struggling?

**Comment by uschi2000** [comment]
Instead of extraDirs, how about allowing to set the baseDir from which
files are copied?

**Comment by mglazer** [comment]
I can share the repo with you offline (it's not a huge deal right now though since we worked around it).

I would also accept a baseDir argument which defaults to CWD, but can be configured to something else (like build/work).

**Comment by mglazer** [comment]
I do think though that this should just do what the standard Gradle Distribution plugin does.
